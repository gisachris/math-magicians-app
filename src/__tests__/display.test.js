import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import HomePage from '../components/home';
import AppHolder from '../components/calculator';
import Quote from '../components/quote';

describe('testing all UI components', () => {
  test('homePage UI testing', () => {
    render(<HomePage />);

    expect(screen.getByText(/Welcome to our Page!/)).toBeInTheDocument();
    expect(screen.getByTestId('homeHolder')).toBeInTheDocument();
  });

  test('calculatorPage UI testing', () => {
    render(<AppHolder />);

    expect(screen.getByTestId('calcWhole')).toBeInTheDocument();
  });
});

describe('QuotesPage UI testing', () => {
  test('renders "quote is loading..." message', () => {
    render(<Quote />);
    expect(screen.queryByText('quote is loading...')).toBeInTheDocument();
  });

  test('renders fetched quotes correctly', async () => {
    const mockResponse = [{ quote: 'To be or not to be', author: 'William Shakespeare' }];
    global.fetch = jest.fn(() => Promise.resolve({ json: () => Promise.resolve(mockResponse) }));
    render(<Quote />);
    await waitFor(() => expect(screen.queryByTestId('quotes')).toBeInTheDocument());
    expect(screen.queryAllByTestId('quotes')).toHaveLength(1);
    expect(screen.queryByText('To be or not to be')).toBeInTheDocument();
    expect(screen.queryByText('ByWilliam Shakespeare')).toBeInTheDocument();
    delete global.fetch;
  });

  test('handles API errors', async () => {
    global.fetch = jest.fn(() => Promise.reject(new Error('API down')));
    render(<Quote />);
    await waitFor(() => expect(screen.queryByTestId('quotes')).toBeNull());
    delete global.fetch;
  });

  test('handles empty quotes', async () => {
    const mockResponse = [];
    global.fetch = jest.fn(() => Promise.resolve({ json: () => Promise.resolve(mockResponse) }));
    render(<Quote />);
    await waitFor(() => expect(screen.queryByTestId('quotes')).toBeNull());
    delete global.fetch;
  });
});
