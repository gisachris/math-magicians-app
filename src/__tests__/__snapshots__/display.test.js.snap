// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Snaphots homePage renders correctly 1`] = `
<section
  className="homeHolder"
  data-testid="homeHolder"
>
  <h1
    className="introTextHeader"
  >
    Welcome to our Page!
  </h1>
  <article
    className="introTextParagragh"
  >
    <h2
      className="parHeader"
    >
      This page is dedicated to creating the best calculator on the world wide web.
    </h2>
    <p
      className="par"
    >
      With this tool, you can perform a wide range of calculations in a fast and user-friendly way. Whether you need to add numbers, subtract them, calculate percentages, or perform more complex operations, our calculator has got you covered
    </p>
    <p
      className="par"
    >
       Designed with simplicity and efficiency in mind, our calculator is perfect for anyone who needs to perform calculations quickly and easily. So, why waste time and effort doing math by hand when you can use our calculator? Start crunching numbers now and see how easy it can be!
    </p>
  </article>
</section>
`;

exports[`QuotePage Snapshots QuotePage renders correctly with no props 1`] = `
<span
  className="noQuo generic"
>
  quote not found
</span>
`;

exports[`QuotePage Snapshots renders correctly with author prop 1`] = `
<span
  className="noQuo generic"
>
  quote not found
</span>
`;

exports[`QuotePage Snapshots renders correctly with both props 1`] = `
<span
  className="loading generic"
>
  quote is loading...
</span>
`;

exports[`QuotePage Snapshots renders correctly with text prop 1`] = `
<span
  className="loading generic"
>
  quote is loading...
</span>
`;

exports[`calculator Snaphots calculator renders correctly 1`] = `
<div
  className="calcWhole"
  data-testid="calcWhole"
>
  <h1
    className="CalcText"
  >
    lets do some math!
  </h1>
  <div
    className="holderBorder"
  >
    <section
      className="holder"
    >
      <section
        className="answer"
      >
        <span
          className="answerSlot"
        />
        <input
          className="input"
          id="input"
          onChange={[Function]}
          placeholder="0"
          type="text"
          value={0}
        />
      </section>
      <section
        className="mainActions"
      >
        <ul
          className="mainKeys"
        >
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            AC
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            +/-
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            %
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            7
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            8
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            9
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            4
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            5
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            6
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            1
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            2
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            3
          </li>
          <li
            className="MainActionTask zeroSpan"
            onClick={[Function]}
          >
            0
          </li>
          <li
            className="MainActionTask"
            onClick={[Function]}
          >
            .
          </li>
        </ul>
      </section>
      <section
        className="SideBar"
      >
        <ul
          className="actionsList"
        >
          <li
            className="actionTask divide"
            onClick={[Function]}
          >
            รท
          </li>
          <li
            className="actionTask multiply"
            onClick={[Function]}
          >
            x
          </li>
          <li
            className="actionTask subtract"
            onClick={[Function]}
          >
            -
          </li>
          <li
            className="actionTask add"
            onClick={[Function]}
          >
            +
          </li>
          <li
            className="actionTask equals"
            onClick={[Function]}
          >
            =
          </li>
        </ul>
      </section>
    </section>
  </div>
</div>
`;
